### PR TITLE
Add make notes command, exclude venv in MANIFEST and gitignore

### PR DESCRIPTION
**Pull request by kclowes**
## What was wrong?
I noticed a few things when I was trying to release v2.x.
- There was no `make notes` command to compile release notes
- venv files weren't being gitignored
- venv files had the potential to make their way into the dist

## How was it fixed?
- Added a few commands to the Makefile.
- added venvs to .gitignore
- added venv files to be excluded in the MANIFEST.in

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/eth-abi/blob/master/newsfragments/README.md)

[//]: # (See: https://eth-abi.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-abi/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Review by pacrob — APPROVED** [review]
sorry for delay, lgtm!

File: Makefile [hunk 1]
```diff
@@ -24,14 +24,14 @@ clean-pyc:
 	find . -name '*~' -exec rm -f {} +
 
 lint:
-	tox -e lint
+	tox -elint
```
Comment (fselmo) [inline Makefile]: Just curious, was there a reason for this change? It seems it still works as `tox -elint` but the usage docs examples separate the environment from the `-e`.

**Review by fselmo — APPROVED** [review]
Just a nit / question on the tox env command change. Also we talked about updating the `Makefile` to match the template project repo since this it has been updated in the time this PR was open (sorry this took a while!).

lgtm though 👍

File: Makefile (hunk 1 above)
Comment (kclowes) [reply Makefile]: Oops, nope. Not sure where this change came from.

**Pull request merged**